src/adapters/voyage/helpers/embedding.ts

```diff
@@ -0,0 +1,25 @@
+import { VoyageAIClient } from "voyageai";
+import { Context } from "../../../types";
+import { SuperVoyage } from "./voyage";
+const VECTOR_SIZE = 1024;
+
+export class Embedding extends SuperVoyage {
+  protected context: Context;
+
+  constructor(client: VoyageAIClient, context: Context) {
+    super(client, context);
+    this.context = context;
+  }
+
+  async createEmbedding(text: string | null): Promise<number[]> {
+    if (text === null) {
+      return new Array(VECTOR_SIZE).fill(0);
+    } else {
+      const response = await this.client.embed({
+        input: text,
+        model: "voyage-2",
```

Comment:
```suggestion
        model: "voyage-large-2-instruct",
```

Should be using the best model. 

> Top of [MTEB leaderboard](https://huggingface.co/spaces/mteb/leaderboard). Instruction-tuned general-purpose embedding model optimized for clustering, classification, and retrieval. For retrieval, please use input_type parameter to specify whether the text is a query or document. For classification and clustering, please use the instructions [here](https://github.com/voyage-ai/voyage-large-2-instruct). See [blog post](https://blog.voyageai.com/2024/05/05/voyage-large-2-instruct-instruction-tuned-and-rank-1-on-mteb/) for details.

[Source](https://docs.voyageai.com/docs/embeddings)